src/ashley/views.py

```diff
@@ -0,0 +1,117 @@
+
+    def post(self, request, *args, **kwargs) -> HttpResponse:  # pylint: disable=W0613
+        """Handler for POST requests."""
+        lti_request = LTI(request)
+        try:
+            lti_request.verify()
+            user = authenticate(request, lti_request=lti_request)
+            if user is not None:
+                login(request, user)
+                return self._do_on_login(lti_request)
+            return HttpResponseForbidden("Forbidden")
+        except LTIException:
+            return HttpResponseForbidden("Forbidden")
+
+    @abstractmethod
+    def _do_on_login(self, lti_request: LTI) -> HttpResponse:
+        """Process the request when the user is logged in via LTI"""
+
+
+class ForumLTIView(BaseLTIView):
+    """Forum view called by an LTI launch request."""
+
+    def _do_on_login(self, lti_request: LTI) -> HttpResponse:
+        """Process the request when the user is logged in via LTI"""
+
+        # Get or create the LTIContext model associated with the current LTI launch request
+        context_id = lti_request.get_param("context_id")
+        if not context_id:
+            return HttpResponseBadRequest()
+        context, _context_created = LTIContext.objects.get_or_create(
+            remote_id=context_id, lti_consumer=lti_request.get_consumer()
+        )
+
+        # Synchronize the user groups related to the current LTI context
+        context.sync_user_groups(self.request.user, lti_request.roles)
+
+        forum_uuid = self.kwargs["uuid"]
+        forum, forum_created = Forum.objects.get_or_create(
+            lti_id=forum_uuid,
+            type=Forum.FORUM_POST,
+            defaults={"name": lti_request.context_title, "type": Forum.FORUM_POST},
+        )
+        if forum_created:
+            logger.debug("Forum %s created", forum_uuid)
+            self._init_forum(forum, context)
+
+        response = HttpResponseRedirect(
+            reverse("forum:forum", kwargs={"slug": forum.slug, "pk": forum.id})
+        )
+
+        locale = lti_request.get_param("launch_presentation_locale")
+        if locale:
+            logger.debug("Course locale detected %s", locale)
+            translation.activate(locale)
+            response.set_cookie(settings.LANGUAGE_COOKIE_NAME, locale)
+
+        return response
+
+    def _init_forum(self, forum, context):
+        """
+        Initialize a forum.
+
+        This method set default permissions according to the LTI context that
+        initiated the creation of the forum.
+        """
+        forum.lti_contexts.add(context)
+        self._assign_perm(
+            forum, context.base_group(), settings.ASHLEY_DEFAULT_FORUM_BASE_PERMISSIONS
+        )
+        for role, perms in settings.ASHLEY_DEFAULT_FORUM_ROLES_PERMISSIONS.items():
+            self._assign_perm(forum, context.role_group(role), perms)
+
+    @staticmethod
+    def _assign_perm(forum, group, permissions: List[str]):
+        for perm in permissions:
```

Comment:
missing docstring. 
I'm surprised the linters didn't break?

Reply:
Fixed :+1:
The linter does not require docstring on protected methods.